packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -534,10 +534,10 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
     protected renderTailDecorations(node: TreeNode, props: NodeProps): React.ReactNode {
         const style = (fontData: TreeDecoration.FontData | undefined) => this.applyFontStyles({}, fontData);
         return <React.Fragment>
-            {this.getDecorationData(node, 'tailDecorations').filter(notEmpty).reduce((acc, current) => acc.concat(current), []).map(decoration => {
+            {this.getDecorationData(node, 'tailDecorations').filter(notEmpty).reduce((acc, current) => acc.concat(current), []).map((decoration, index) => {
                 const { fontData, data, tooltip } = decoration;
                 const className = [TREE_NODE_SEGMENT_CLASS, TREE_NODE_TAIL_CLASS].join(' ');
-                return <div key={node.id + className} className={className} style={style(fontData)} title={tooltip}>
+                return <div key={node.id + className + index} className={className} style={style(fontData)} title={tooltip}>
```

Comment:
I wonder whether `index` should be sufficient, `key` should be unique only among siblings not globally.

Reply:
@akosyakov thanks for suggesting using index, which was my possible alternative solution 3.

Yes, but the siblings include other decorators types (caption prefixes, suffixes etc) as React.Fragment does not generate an element.  This would assume nothing else used only numbers as keys.  It does seem to me that using class names as keys is not a great pattern, but it was the one used.

The use of index does mean we have unstable keys, which does not matter unless extenders make custom changes (not using react) to their decorators.  It seems you are fine with that.

Reply:
ok, fine with me, merging?